openwisp_radius/api/freeradius_views.py

```diff
@@ -428,7 +433,26 @@ def post(self, request, *args, **kwargs):
             instance = self.get_queryset().get(unique_id=data.get('unique_id'))
         except RadiusAccounting.DoesNotExist:
             serializer = self.get_serializer(data=data)
-            serializer.is_valid(raise_exception=True)
+            try:
+                serializer.is_valid(raise_exception=True)
+            except ValidationError as error:
+                # Handles "Interim-Updates" for RadiusAccounting sessions
+                # that are closed by OpenWISP when user logs into
+                # another organization.
+                unique_id_errors = error.detail.get('unique_id', [])
+                if len(unique_id_errors) == 1:
+                    error_detail = unique_id_errors.pop()
+                    if (
+                        str(error_detail)
+                        == 'accounting with this accounting unique ID already exists.'
+                        and error_detail.code == 'unique'
+                    ):
+                        rad = RadiusAccounting.objects.only(
+                            'unique_id', 'organization_id'
+                        ).get(unique_id=data.get('unique_id'))
+                        if rad.organization_id != request.auth:
+                            return Response(None)
```

Comment:
can you please make this whole block a private method of the view called `_is_interim_update_corner_case()` and then just do:

```python
if self._is_interim_update_corner_case(error):
    return Response(None)
raise error
```

This way we keep the `post` method leaner, it's always the best thing to maintain this kind of complex logic, especially for corner cases, separate.